### DISCUSSION TITLE
How to change the background color of `Canvas` associated to `ItemsControl` using C# Markup?

### DISCUSSION THREAD

**Discussion by kucint** [unoplatform/uno#16091] [Q&A] [answered]
I want to change the background color of `Canvas` associated to `ItemsControl`:
To do so I produced following code:

```
public sealed partial class MainPage : Page
{
    public MainPage() => this
            .Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"))
            .Content(
              new ItemsControl()
              .ItemsPanel(
                  () => new Canvas()
                  // uncommenting any of the following lines will cause the app to crash
                  //.Width(250)
                  //.Height(250)
                  //.Background(Colors.DarkBlue)
                  ));
}
```

Unfortunately, this doesn't work, it causes app to crash:

_Exception thrown: 'System.ArgumentException' in WinRT.Runtime.dll
An exception of type 'System.ArgumentException' occurred in WinRT.Runtime.dll but was not handled in user code_

The equivalent Code in XAML would be:
```
      <ItemsPanelTemplate>
        <Canvas Background="Transparent"/>
      </ItemsPanelTemplate>
    </ItemsControl.ItemsPanel>
```

BTW: modifying `Width` or `Height` of a `Canvas` causes the crash as well.

How do I realize it in C# Markup?

See attached uno project that reproduces this issue:
[UnoCanvasApp.16091.zip](https://github.com/unoplatform/uno/files/14837176/UnoCanvasApp.16091.zip)

**Comment by dansiegel** [comment, marked as answer]
Thanks @kucint this would actually be the result of a bug we recently discovered. The team is currently working to resolve it. The fix should be in the 5.2 release. Note that to properly fix this we will need to make a breaking change that is specific to the ItemsPanelTemplate. The updated code will look something like this:

```cs
new ItemsControl()
    .ItemsPanel<Canvas>(canvas => canvas
        .Width(250)
        .Height(250)
        .Background(Colors.Blue))
```

cc: @kazo0 @eriklimakc @vatsashah45
